src/common/freeze/deepfreeze.ts

```diff
@@ -0,0 +1,21 @@
+
+export const deepFreezeSkipDate = (o) => {
```

Comment:
Have you considered saving dates as int? Sounds radical, but it simplifies state, and Date object can be created only when needed. It can be compared and sorted as int.
After thinking a bit on it, I got convinced, and it may be easy to apply, because if somebody sneaks a Date in a reducer state, it will make deepFreeze to fail (correct me if I'm wrong).